File: all/abc327/abc327_b/abc327_b.cs
class A
{
  static long pow(long a, long b)
  {
    long res = 1;
    while(b > 0)
    {
      b--;
      res *= a;
    }

    return res;
  }

  static void Main()
  {
    long b = long.Parse(Console.ReadLine());
    long max = 10000000;

    for(long i = 1; i < max; i++)
    {
      if(pow(i, i) > b) break;
      if(pow(i, i) == b)
      {
        Console.Write(i);
        return;
      }
    }

    Console.Write(-1);
  }
}

File: abc/abc327/abc327_b/abc327_b.py
def pow(x, y):
  res = 1
  for i in range(y):
    res *= x

  return res

b = int(input())
for i in range(1, 1000000):
  r = pow(i, i)
  if r > b:
    break

  if r == b:
    print(i)
    exit()

print(-1)